### PR TITLE
Fix Error Prone warnings

### PR DESCRIPTION
**Pull request by brijeshshah13**
Fixes, in continuation to #989.

Changes: Solved and suppressed warnings and errors.

Screenshots for the change: N/A.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
file:///home/circleci/code/app/src/main/res/layout/session_layout.xml Line:23
/home/circleci/code/app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java:51: error: [CheckReturnValue] Ignored return value of method that is annotated with @CheckReturnValue
            .subscribe(createdTrack -> {
                      ^
    (see http://errorprone.info/bugpattern/CheckReturnValue)
  Did you mean to remove this line?
1 error
2 warnings
:app:compileDebugJavaWithJavac FAILED

File: app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ private void nullifyEmptyFields(Track track) {
         track.setDescription(StringUtils.emptyToNull(track.getDescription()));
     }
 
+    @SuppressWarnings("CheckReturnValue")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/track/create/CreateTrackPresenter.java]: You should fix this rather than suppressing

File: app/src/main/java/org/fossasia/openevent/app/data/Repository.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 /**
  * General Repository class. To be generified in future
  */
+@SuppressWarnings("MissingOverride")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/Repository.java]: Add override rather than suppressing

File: app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeRepositoryImpl.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public AttendeeRepositoryImpl(Repository repository, AttendeeApi attendeeApi) {
             .subscribeOn(Schedulers.io());
     }
 
-    @NonNull
+    @Override @NonNull
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeRepositoryImpl.java]: Add in new line

File: app/src/main/java/org/fossasia/openevent/app/data/db/DbFlowDatabaseChangeListener.java [hunk 4]
```diff
@@ -53,6 +55,7 @@ public void onModelChanged(@NonNull T model, @NonNull BaseModel.Action action) {
         DirectModelNotifier.get().registerForModelChanges(classType, modelModelChangedListener);
     }
 
+    @SuppressWarnings("MissingOverride")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/db/DbFlowDatabaseChangeListener.java]: Add overrides

File: app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java [hunk 5]
```diff
@@ -67,4 +68,8 @@
     @Relationship("user")
     @ForeignKey(stubbedRelationship = true, onDelete = ForeignKeyAction.CASCADE)
     public User user;
+
+    @Relationship("session")
+    @ForeignKey(stubbedRelationship = true, onDelete = ForeignKeyAction.CASCADE)
+    public Session session;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java]: Nope, remove this

File: app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java [hunk 6]
```diff
@@ -26,7 +27,7 @@
 @Type("speaker")
 @AllArgsConstructor
 @NoArgsConstructor
-@ToString(exclude = {"event", "sessions", "user"})
+@ToString(exclude = {"event", "session", "user"})
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java]: Remove session from here

Comment (brijeshshah13) [reply app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java]: Okay

File: app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java (hunk 5 above)
Comment (brijeshshah13) [reply app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java]: Yeah, that follows the removal of session

File: app/src/main/java/org/fossasia/openevent/app/data/db/DbFlowDatabaseChangeListener.java (hunk 4 above)
Comment (brijeshshah13) [reply app/src/main/java/org/fossasia/openevent/app/data/db/DbFlowDatabaseChangeListener.java]: Alright

File: app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeRepositoryImpl.java (hunk 3 above)
Comment (brijeshshah13) [reply app/src/main/java/org/fossasia/openevent/app/data/attendee/AttendeeRepositoryImpl.java]: Np

**Comment by brijeshshah13** [comment]
@iamareebjamal Btw, do checkout the current circleci build. It displays no errors and only warnings, still the build is failing.

File: app/src/main/java/org/fossasia/openevent/app/data/Repository.java (hunk 2 above)
Comment (brijeshshah13) [reply app/src/main/java/org/fossasia/openevent/app/data/Repository.java]: @iamareebjamal Replacing it with override shows this error: `'@Override' not applicable to type`

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/app/data/Repository.java]: OK

**Comment by brijeshshah13** [comment]
@iamareebjamal Please guide me on how what is failing the builds now.

File: app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java [hunk 7]
```diff
@@ -13,6 +13,7 @@
 
 import org.fossasia.openevent.app.data.db.configuration.OrgaDatabase;
 import org.fossasia.openevent.app.data.event.Event;
+import org.fossasia.openevent.app.data.session.Session;
```
[the remark this answers is not in the document]
Comment (brijeshshah13) [reply app/src/main/java/org/fossasia/openevent/app/data/speaker/Speaker.java]: Removed

**Comment by iamareebjamal** [comment]
https://github.com/google/error-prone/issues/771

**Comment by brijeshshah13** [comment]
@iamareebjamal I tried almost everything from that link to fix it without success. I request you to have a look at it. @codedsun Can you see if you can fix it?

**Comment by codedsun** [comment]
@brijeshshah13 I am still keeping my head since the last night, and I m encountering the same error


```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:compileDebugJavaWithJavac'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
	at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:59)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:256)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:249)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:238)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:663)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:597)
	at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed with exit code 1; see the compiler error output for details.
	at net.ltgt.gradle.errorprone.ErrorProneCompiler.execute(ErrorProneCompiler.java:80)
	at net.ltgt.gradle.errorprone.ErrorProneCompiler.execute(ErrorProneCompiler.java:24)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:98)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:51)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:37)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:35)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:25)
	at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:156)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:141)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:114)
	at com.android.build.gradle.tasks.factory.AndroidJavaCompile.compile(AndroidJavaCompile.java:125)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
	at org.gradle.api.internal.project.taskfactory.IncrementalTaskAction.doExecute(IncrementalTaskAction.java:50)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
	... 33 more

```

**Comment by codedsun** [comment]
@brijeshshah13 When I checked more of it, it seems there is an issue while integrating the JDK8 and I applied all the solutions which were mentioned in the Issues section but still not able to resolve @iamareebjamal Please guide!

**Comment by brijeshshah13** [comment]
@codedsun You must have got compile time errors. Checkout the `Build` section to view what errors you are getting.

**Comment by codedsun** [comment]
@brijeshshah13 I am getting some of the warnings which can be supressed and I did supressed that, but no effect on this error and by the build-log of your PR says the same. Correct me where I am getting wrong!

**Comment by iamareebjamal** [comment]
@brijeshshah13 As the error says, ParameterName is invalid flag, just remove it

File: app/config/checkstyle.xml [hunk 8]
```diff
@@ -73,7 +73,6 @@ Severity: HARD
         <module name="MemberName" />
         <module name="MethodName" />
         <module name="PackageName" />
-        <module name="ParameterName" />
```
Comment (iamareebjamal) [inline app/config/checkstyle.xml]: Since this is unrelated to the error prone integration, this is not where the problem lies. See the changelist in the commit where ParameterName label has been added

Comment (brijeshshah13) [reply app/config/checkstyle.xml]: @iamareebjamal You mean in the past commits?

**Comment by brijeshshah13** [comment]
@iamareebjamal The

**Comment by iamareebjamal** [comment]
You removed it correctly now @brijeshshah13 Now the error is from Dagger Annotations

**Comment by brijeshshah13** [comment]
@iamareebjamal But to the best of my knowledge, classes like `DaggerAppComponent` are created while building, so how can we solve that?

**Comment by iamareebjamal** [comment]
This is due to errorprone version mismatch

**Comment by brijeshshah13** [comment]
@iamareebjamal Which version shall I change it to? Because upgrading to the latest version 2.3.1 again throws the same issue: https://github.com/google/error-prone/issues/780.

**Comment by iamareebjamal** [comment]
I'm looking into it

**Comment by brijeshshah13** [comment]
@iamareebjamal Alright. Meanwhile, I will work on other issues.
